### PR TITLE
Change all scheduleAtFixedRate method references to method scheduleWi…

### PR DESCRIPTION
**Pull request by lpapazow**
Regarding [this issue](https://github.com/openhab/static-code-analysis/issues/62) all scheduleAtFixedRate method references have been replaced with scheduleWithFixedDelay method references. 

Additionally, a check is implemented in the Static Code Analysis Tool.

### CODE REVIEW

File: addons/binding/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/handler/HDPowerViewHubHandler.java [hunk 1]
```diff
@@ -123,7 +123,7 @@ private void schedulePoll() {
             pollFuture.cancel(false);
         }
         logger.debug("Scheduling poll for 500ms out, then every {} ms", refreshInterval);
-        pollFuture = scheduler.scheduleAtFixedRate(pollingRunnable, 500, refreshInterval, TimeUnit.MILLISECONDS);
+        pollFuture = scheduler.scheduleWithFixedDelay(pollingRunnable, 500, refreshInterval, TimeUnit.MILLISECONDS);
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/handler/HDPowerViewHubHandler.java]: Please do not yet touch the MILLISECONDS in most cases there are some stricter timings needed in those. Can you check others as well?

Comment (kaikreuzer) [reply addons/binding/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/handler/HDPowerViewHubHandler.java]: @martinvw What exactly do you mean by this? That anything that is scheduled more frequently than once a second should be done at fixed rate and not with fixed delay?

Comment (martinvw) [reply addons/binding/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/handler/HDPowerViewHubHandler.java]: Not completely, but everything that is scheduled so often might have a valid use case. In all others minute-resolution etc I'm sure that it does not matter.

Comment (martinvw) [reply addons/binding/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/handler/HDPowerViewHubHandler.java]: Note that in this specific case the default is set 60 000 so it does not matter, but there is for example some fading logic (I believe in the WIFI led binding) which might break by changing it to a fixedDelay.

Comment (kaikreuzer) [reply addons/binding/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/handler/HDPowerViewHubHandler.java]: Well, especially these cases are risky as if the job takes too long, you might end up running multple in parallel - and when debugging and stepping through your code, the fixed rate is pretty nasty as well.
So unless there is some requirement that it has to be exactly 500ms, I would vote for changing it - and imho that should be ok for almost all places in the code.

Comment (kaikreuzer) [reply addons/binding/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/handler/HDPowerViewHubHandler.java]: > some fading logic

Yes, you are right, that would be a valid example where we'd need to keep the fixedRate.

Comment (lpapazow) [reply addons/binding/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/handler/HDPowerViewHubHandler.java]: @kaikreuzer  , @martinvw 

Could you please specify all the cases in which scheduleAtFixedRate should not be changed, or  tell me how to determine if I should not change them.

Additionally, I could add a white list of allowed use cases for scheduleAtFixedRate calls in the checkstyle rule.

File: addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/FadingWiFiLEDDriver.java [hunk 2]
```diff
@@ -301,7 +301,7 @@ synchronized private void fadeToState(final InternalLedState newTargetState) thr
                     });
             ledfaderThread.init();
             final int period = fadeDurationInMs / totalFadingSteps;
-            final Future<?> future = faderExecutor.scheduleAtFixedRate(ledfaderThread, 0, period < 1 ? 1 : period, TimeUnit.MILLISECONDS);
+            final Future<?> future = faderExecutor.scheduleWithFixedDelay(ledfaderThread, 0, period < 1 ? 1 : period, TimeUnit.MILLISECONDS);
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/FadingWiFiLEDDriver.java]: Should not be changed

File: addons/binding/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/handler/HDPowerViewHubHandler.java (hunk 1 above)
Comment (martinvw) [reply addons/binding/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/handler/HDPowerViewHubHandler.java]: I marked the one I could find the rest should be safe to change imho

**Comment by J-N-K** [comment]
DMX should not be changed either, most receivers for DMX-over-Ethernet rely on constant-rate-packets.

**Comment by martinvw** [comment]
You added commits which do not belong here, maybe you should strip of some commits `git reset HEAD~{nr-of-commits} --hard` and redo the reverting of the changes to the `FadingWifiLEDDriver`

**Comment by martinvw** [comment]
Thanks!

**Pull request merged**